### PR TITLE
Added documentation for copying a Map

### LINKED ISSUE #4259: Document how to copy a map
Would be nice to be able to create copies of maps, e.g. sometimes I want to modify a map, but also keep a copy of the original hanging round. In my head this is just a case of copying the data and header and putting it in a new `Map`, but I'm sure it's more complicated (and there are other considerations as to whether this should be implemented)

### PR DESCRIPTION
**Pull request by chouhanaryan**
### Description
- Added documention for copying [Map](https://docs.sunpy.org/en/stable/guide/data_types/maps.html) objects.
- Fixed a very small typo in the same section.

Fixes #4259

### CODE REVIEW

**Comment by chouhanaryan** [comment]
@nabobalis 
There are two other things I'd like to clarify:
- When I ran `pytest`, I did get this error as well `FAILED guide\acquiring_data\fido.rst::fido.rst`. I haven't touched this file so I'm assuming this error already exists. If there's anything specific I'm missing, please let me know.
- The CI pipeline mentions adding something to the changelog. It's a pretty small change so does it warrant an update?

I'm assuming the other failures (codecov) are already present as well?

**Comment by nabobalis** [comment]
> * When I ran `pytest`, I did get this error as well `FAILED guide\acquiring_data\fido.rst::fido.rst`. I haven't touched this file so I'm assuming this error already exists. If there's anything specific I'm missing, please let me know.

What error? It could be due to column widths than a real problem. 

> * The CI pipeline mentions adding something to the changelog. It's a pretty small change so does it warrant an update?

Yes, this should get a documentation fix changelog.
 
> I'm assuming the other failures (codecov) are already present as well?

Yeah.

File: docs/guide/data_types/maps.rst [hunk 1]
```diff
@@ -283,6 +283,17 @@ elsewhere, you can use either of the following::
 Python makes use of pointers so if you want to alter the data and keep the
 original data in the map intact make sure to copy it.
 
+To create a complete copy of a Map object that is entirely independent of the original,
+use the built-in `copy https://docs.python.org/3/library/copy.html`_ module's deepcopy
```
Comment (nabobalis) [inline docs/guide/data_types/maps.rst:287]:
```suggestion
use the built-in `copy https://docs.python.org/3/library/copy.html`__ module's deepcopy
```

**Comment by chouhanaryan** [comment]
@nabobalis 
The `fido.rst` error is as follows -

```
Aryan D:\sunpy-git\docs git: document-map-copy-4259 ≣ env:venv ❯❯❯ pytest
=========================================================== test session starts ===========================================================
platform win32 -- Python 3.7.4, pytest-6.0.2, py-1.9.0, pluggy-0.13.1
rootdir: D:\sunpy-git, configfile: setup.cfg
plugins: asdf-2.7.1, hypothesis-5.35.0, arraydiff-0.3, astropy-header-0.1.2, cov-2.10.1, doctestplus-0.8.0, filter-subpackage-0.1.1, mock-3.3.1, openfiles-0.5.0, remotedata-0.3.2
collected 28 items

code_ref\io.rst .                                                                                                                    [  3%]
code_ref\map.rst s                                                                                                                   [  7%]
code_ref\net.rst .                                                                                                                   [ 10%]
code_ref\timeseries.rst .                                                                                                            [ 14%]
code_ref\coordinates\index.rst .                                                                                                     [ 17%]
code_ref\coordinates\rotatedsunframe.rst .                                                                                           [ 21%]
code_ref\coordinates\wcs.rst .                                                                                                       [ 25%]
dev_guide\logger.rst .                                                                                                               [ 28%]
dev_guide\units_quantities.rst s                                                                                                     [ 32%]
guide\customization.rst .                                                                                                            [ 35%]
guide\logger.rst .                                                                                                                   [ 39%]
guide\plotting.rst s                                                                                                                 [ 42%]
guide\roi.rst .                                                                                                                      [ 46%]
guide\time.rst .                                                                                                                     [ 50%]
guide\tour.rst .                                                                                                                     [ 53%]
guide\troubleshooting.rst s                                                                                                          [ 57%]
guide\units-coordinates.rst .                                                                                                        [ 60%]
guide\acquiring_data\database.rst s                                                                                                  [ 64%]
guide\acquiring_data\fido.rst F                                                                                                      [ 67%]
guide\acquiring_data\hek.rst .                                                                                                       [ 71%]
guide\acquiring_data\helioviewer.rst .                                                                                               [ 75%]
guide\acquiring_data\jsoc.rst .                                                                                                      [ 78%]
guide\data_types\maps.rst .                                                                                                          [ 82%]
guide\data_types\timeseries.rst .                                                                                                    [ 85%] 
whatsnew\0.8.rst s                                                                                                                   [ 89%]
whatsnew\0.9.rst s                                                                                                                   [ 92%]
whatsnew\1.0.rst s                                                                                                                   [ 96%]
whatsnew\2.0.rst s                                                                                                                   [100%]

================================================================ FAILURES ================================================================= 
___________________________________________________________ [doctest] fido.rst ____________________________________________________________ 
039     (<sunpy.net.attrs.Time(2012-03-04 00:00:00.000, 2012-03-06 00:00:00.000)>, <sunpy.net.attrs.Instrument(LYRA: Lyman Alpha Radiometer 
is the solar UV radiometer on board
040     Proba-2.) object at ...>)
041
042 For attributes that have no fixed selection of values (``Time`` for example) you will have to provide the range you require.
043 However, for attributes that have a fixed range of **known** values, it is possible to list all these values.
044 **Please note that each list is not exhaustive.**
045
046 Using ``Instrument`` as the first example, if you print the object::
047
048     >>> print(a.Instrument)
Differences (unified diff with -expected +actual):
    @@ -3,10 +3,104 @@
     Specifies the Instrument name for the search.
     <BLANKLINE>
    -       Attribute Name          Client          Full Name                                           Description
    ---------------------------- ----------- ------------------------ --------------------------------------------------------------------------------
    -aia                         VSO         AIA                      Atmospheric Imaging Assembly
    -bbi                         VSO         BBI                      None
    -bcs                         VSO         BCS                      Bragg Crystal Spectrometer
    -bic_hifi                    VSO         BIC-HIFI                 None
    -bigbear                     VSO         Big Bear                 Big Bear Solar Observatory, California TON and GONG+ sites
    -...
    +       Attribute Name       ...
    +--------------------------- ...
    +aia                         ...
    +bbi                         ...
    +bcs                         ...
    +bic_hifi                    ...
    +bigbear                     ...
    +caii                        ...
    +cds                         ...
    +celias                      ...
    +cerrotololo                 ...
    +chp                         ...
    +chrotel                     ...
    +climso                      ...
    +costep                      ...
    +cp                          ...
    +dpm                         ...
    +eis                         ...
    +eit                         ...
    +elteide                     ...
    +erne                        ...
    +eve                         ...
    +eve                         ...
    +five_12_channelmagnetograph ...
    +foxsi                       ...
    +gbm                         ...
    +gfpi                        ...
    +goes                        ...
    +golf                        ...
    +gris                        ...
    +ha2                         ...
    +hi_c                        ...
    +hi_c21                      ...
    +hmi                         ...
    +hxeclipse                   ...
    +hxt                         ...
    +imax                        ...
    +impact                      ...
    +iris                        ...
    +isoon                       ...
    +iss                         ...
    +ivm                         ...
    +k_cor                       ...
    +kpdc                        ...
    +lasco                       ...
    +learmonth                   ...
    +longwave_lobe_06            ...
    +longwave_lobe_07            ...
    +longwave_slit_06            ...
    +longwave_slit_07            ...
    +lyra                        ...
    +maunaloa                    ...
    +mdi                         ...
    +mergedgong                  ...
    +meudonspectroheliograph     ...
    +mk4                         ...
    +noaa_indices                ...
    +noaa_predict                ...
    +norh                        ...
    +ovsa                        ...
    +phoenix                     ...
    +plastic                     ...
    +ptmc                        ...
    +rhessi                      ...
    +rhessi                      ...
    +secchi                      ...
    +shortwave_lobe_06           ...
    +shortwave_lobe_07           ...
    +shortwave_slit_06           ...
    +shortwave_slit_07           ...
    +six_0_ftshg                 ...
    +sj                          ...
    +solarftsspectrometer        ...
    +soon                        ...
    +sot                         ...
    +sp1                         ...
    +sp2                         ...
    +spectroheliograph           ...
    +spectromagnetograph         ...
    +srs_table                   ...
    +sumer                       ...
    +suvi                        ...
    +swan                        ...
    +swap                        ...
    +swaves                      ...
    +sxi_0                       ...
    +sxt                         ...
    +tm_1001                     ...
    +tm_1010                     ...
    +trace                       ...
    +udaipur                     ...
    +uvcs                        ...
    +vault_1999                  ...
    +vault_2002                  ...
    +vault_2014                  ...
    +virgo                       ...
    +vsm                         ...
    +wbs                         ...
    +wispr                       ...
    +xrs                         ...
    +xrt                         ...
    +zimpol                      ...

D:\sunpy-git\docs\guide\acquiring_data\fido.rst:48: DocTestFailure
========================================================= short test summary info ========================================================= 
FAILED guide\acquiring_data\fido.rst::fido.rst
================================================ 1 failed, 18 passed, 9 skipped in 12.69s ================================================= 
```

**Comment by nabobalis** [comment]
That looks like a column width issue, we normally run them with 180 column width which by default pytest does not use.

File: docs/guide/data_types/maps.rst [hunk 2]
```diff
@@ -283,6 +283,17 @@ elsewhere, you can use either of the following::
 Python makes use of pointers so if you want to alter the data and keep the
 original data in the map intact make sure to copy it.
 
+To create a complete copy of a Map object that is entirely independent of the original,
+use the built-in `copy https://docs.python.org/3/library/copy.html`__ module's deepcopy
```
Comment (wtbarnes) [inline docs/guide/data_types/maps.rst:287]: Can we use intersphinx mapping to link to the docs for the `copy` module in the standard library? I think something like

```suggestion
use the built-in `copy.deepcopy`
```

should work.

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks @chouhanaryan

**Pull request merged**